### DISCUSSION TITLE
Use https backend server redirect to http problem

### DISCUSSION THREAD

**Discussion by huang-x-h** [apache/apisix#6802] [Q&A]
When we use https to access apisix, backend server use http.

Now, when we access `https://example.com:9443/path/login.html`, this page will be form submit and the backend server will redirect to '/path/index.html`, then browser url change to `http://example.com:9443/path/index.html`, not `https` protocol

So we add follow snippet, it's work fine.
```
proxy_redirect http://exmaple.com:9443/  https://exmaple.com:9443/;
```

The backend server is Java, we use `HttpServletResponse response.sendRedirect(redirectUrl)`

But i think this is not a right way, have an another way to fixed this?

**Comment by tokers** [comment]
@huang-x-h Did you use the [redirect](https://apisix.apache.org/docs/apisix/plugins/redirect/) plugin? What does the redirect response look like (from the backend).

**Reply by huang-x-h** [reply]
We use version `2.10`,  change the `cli/ngx_tpl.lua` file 😢 

Backend server just use `response.sendRedirect('/path/index.html')`, the browser redirect to `http://exmaple.com:9443/path/index.html`, but we hope redirect to `https://exmaple.com:9443/path/index.html`. So we add `proxy_redirect` to change the protocol

**Reply by tzssangglass** [reply]
we can use `http_to_https` in `redirect` plugins?

**Reply by tokers** [reply]
> redirect

If the backend can split this part of logic to APISIX, I think so, otherwise it won't be effective.

**Reply by tokers** [reply]
@huang-x-h Could you evaluate the proposal from @tzssangglass ?

**Reply by huang-x-h** [reply]
> @huang-x-h Could you evaluate the proposal from @tzssangglass ?

Maybe is our backend code error, when we redirect the url, the response header `location` is `http://exmaple.com:9443/path/index.html`, so we use `proxy_redirect` to change `http` to `https` 😞

**Reply by tokers** [reply]
If so, what about using the [serverless plugin](https://apisix.apache.org/docs/apisix/plugins/serverless/), injecting some codes in the header_filter phase, rewriting the Location header.

**Reply by huang-x-h** [reply]
yes, it's works. Thanks 👍

**Reply by Padmabishek** [reply]
@huang-x-h  Can you please help in understanding how to use the plugin [serverless plugin](https://apisix.apache.org/docs/apisix/plugins/serverless/) for the same? I was facing the same issue when I upgraded my apisix installation to 3.15

**Reply by huang-x-h** [reply]
@Padmabishek we use `http server location configuration snippet`, setting like this:

```
http server location configuration snippet: |
  proxy redirect http: //access_ip:9443 https: //access_ip:9443:
```
